src/UI/Component/Chart/Gauge/Gauge.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/* Copyright (c) 2017 Ralph Dittrich <dittrich@qualitus.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\Chart\Gauge;
+use ILIAS\UI\Component\Component;
+
+/**
+ * Interface Gauge
+ * @package ILIAS\UI\Component\Chart\Gauge
+ */
+interface Gauge extends Component {
+
+    /**
+     * Get maximum value
+     *
+     * This value is used as 100%.
+     * This value will always returned "raw" because it is used to calculate the
+     * percentage values of main, required and comparision.
+     *
+     * @return int|float
+     */
+    public function getMaximum();
+
+    /**
+     * Get required value
+     *
+     * This value represents the required amount that is needed, to fulfill the objective.
+     * If this value is not set, it defaults to the maximum.
+     *
+     * @return int|float
+     */
+    public function getRequired();
+
+    /**
+     * Get main value
+     *
+     * This value is represented as the main gauge bar.
+     *
+     * @return int|float
+     */
+    public function getMainValue();
+
+    /**
+     * Test if a comparison value is given
+     *
+     * This test is used to decide whether one or two bars are rendered.
+     *
+     * @return bool
+     */
+    public function hasComparisonValue();

```

Comment:
I still don't think this method makes sense in that way here. It's super nitpicky, tough.

I mean: look at this interface as found here. By the means provided by it, I could only tell, that some object implementing that interface *has* a comparison value but I cannot know *what that value is*, because the interface provides no method to get that value.

On the other hand, the interface `Standard` defines a `getComparison` method, but does not define what happens when I call it in the case that `hasComparisonValue() == false`. I also cannot know that I can call this method on some `Gauge`-object, even if `hasComparisonValue() == true`. I would need an explicit check on `instanceof Standard` on the object to know the method really exists.

So in fact `hasComparisonValue` provides only very little value to users of the `Gauge`-interface. The really useful check is `instanceof Standard`. On the other hand I could build implementations of `Standard` that show contradictory behaviour regarding `hasComparisonValue` and `getComparisonValue`.

Reply:
I see what you mean. I will look again at this function. Maybe I will throw it away and just test in the renderer if the comparision value is not null.